### ISSUE TITLE
Publish on F-Droid

### ISSUE THREAD

**Issue by mmatous** [Criptext/Android-Email-Client#477]
**Is your feature request related to a problem? Please describe.**
You target privacy-centric crowd but do not provide any other options for app distribution other than GPlay with build dependent on their proprietary solutions.

**Describe the solution you'd like**
I would like to see FOSS build packaged for and published via F-Droid. [How to.](https://f-droid.org/docs/FAQ_-_App_Developers/?title=FAQ_-_App_Developers#how-do-i-get-my-app-included) At minimum your FireBase and Crashlytics would have to be excluded for this build.

**Describe alternatives you've considered**
You could also host your own repo. [How to.](https://f-droid.org/docs/FAQ_-_App_Developers/?title=FAQ_-_App_Developers#can-i-run-my-own-f-droid-package-repository)

**Additional context**
No.

**Comment by jorgeblacio** [comment]
Hi @mmatous, is in our plans in the future once we come out from beta to start exploring options to separate Criptext from things like firebase, because we know the more independent fro this kind of services we are the better. I can't promise an ETA yet, but rest assured that is in our plans.

**Comment by c33s** [comment]
@jorgeblacio maybe you can push this topic a little bit. people who are very privacy aware and want encryption also don't want to use google. so having a privacy tool like yours on f-droid is nearly a must.

**Comment by quantizzed** [comment]
You could probably start by publishing APKs here on GitHub as the first step. There's currently no way of downloading Criptext other than from Google.

Also, if you plan to introduce paid "Pro" features, you need to think of a way they can be purchased without having a Google account. A good example of this is [FairEmail](https://github.com/M66B/FairEmail).

**Comment by licaon-kter** [comment]
https://gitlab.com/fdroid/rfp/-/issues/1337

> All our code is open source, which makes it easy for you to verify that our software works as we say it does.

Which is not actually true, right? We can't actually verify the resulting app since Firebase, Google Services, Crashlytics and maybe other deps are not opensource.

**Comment by jorgeblacio** [comment]
> @jorgeblacio maybe you can push this topic a little bit. people who are very privacy aware and want encryption also don't want to use google. so having a privacy tool like yours on f-droid is nearly a must.

Thanks for the interest, and yes we are aware of the importance of this, which is why I promise I will push this issue up on our next month's feature pipeline planning meeting.

**Comment by jorgeblacio** [comment]
> @jorgeblacio maybe you can push this topic a little bit. people who are very privacy aware and want encryption also don't want to use google. so having a privacy tool like yours on f-droid is nearly a must.

@c33s Thanks for the interest, and yes we are aware of the importance of this, which is why I promise I will push this issue up on our next month's feature pipeline planning meeting.

**Comment by jorgeblacio** [comment]
> You could probably start by publishing APKs here on GitHub as the first step. There's currently no way of downloading Criptext other than from Google.
> 
> 
> 
> Also, if you plan to introduce paid "Pro" features, you need to think of a way they can be purchased without having a Google account. A good example of this is [FairEmail](https://github.com/M66B/FairEmail).

@quantizzed you are right, Starting next release I will update, we will start uploading the corresponding apk. Although it will still have the dependencies it currently has.

**Comment by jorgeblacio** [comment]
> https://gitlab.com/fdroid/rfp/-/issues/1337
> 
> 
> 
> > All our code is open source, which makes it easy for you to verify that our software works as we say it does.
> 
> 
> 
> Which is not actually true, right? We can't actually verify the resulting app since Firebase, Google Services, Crashlytics and maybe other deps are not opensource.


@licaon-kter well technically it is true, since all the code that handles the encryption and decryption and all the signal protocol handling is done on our app and the code to that is freely visible, as well as all the handling of the user data, which again is only handled on the device, and that can be confirmed as well through our open source code. True, you cannot verify the dependencies like fire base code, but we only use fire base and google services for push notifications and crash reports. And the push notification doesn’t have any important data(that can be checked on our code as well), it only delivers an ID to the device, which then the device uses to ask for the encrypted email event on the server, so basically we just use it to let the device know there is something new and nothing else(again this can be checked on our code).

I know is not ideal, but that helped us publish the app way quicker, and rest assured we are working to implement on our server and on our apps open source replacement for this notification services but it will take some time and once that is in place we can finally remove all google dependencies and publish on f-droid.

**Comment by LibreFoss** [comment]
Any news about fdroid?

**Comment by kquote03** [comment]
Hello, has there been any advancements on this?

**Comment by heberjeur** [comment]
Dead projet ?

**Comment by RasheedAZ** [comment]
> Dead projet ?

seems like it's not being developed any longer :(
